x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/schema.ts

```diff
@@ -93,6 +93,24 @@ const numberOfShardsField = {
   label: i18n.translate('xpack.indexLifecycleMgmt.shrink.numberOfPrimaryShardsLabel', {
     defaultMessage: 'Number of primary shards',
   }),
+  defaultValue: 1,
+  validations: [
+    {
+      validator: emptyField(i18nTexts.editPolicy.errors.numberRequired),
+    },
+    {
+      validator: numberGreaterThanField({
+        message: i18nTexts.editPolicy.errors.numberGreatThan0Required,
+        than: 0,
+      }),
+    },
+  ],
+  serializer: serializers.stringToNumber,
+};
+const shardSizeField = {
+  label: i18n.translate('xpack.indexLifecycleMgmt.shrink.primaryShardSizeLabel', {
+    defaultMessage: 'Maximum primary shard size',
```

Comment:
We can omit "primary" here since replicas should be the same size as their primary.
```suggestion
    defaultMessage: 'Maximum shard size',
```